src/sync-merkleroots.test.ts

```diff
@@ -0,0 +1,315 @@
+import fetchMock from 'jest-fetch-mock';
+import { SpvWalletClient } from './client';
+import { ExclusiveStartKeyPage, MerkleRoot, MerkleRootsRepository } from './types';
+import { ErrorStaleLastEvaluatedKey, ErrorSyncMerkleRootsTimeout } from './errors';
+
+type MockedApiResponseType = 'delayed' | 'stale' | 'normal';
+const server = 'http://localhost:3003/api/v1';
+
+// creates a simulated repository a client passes to SyncMerkleRoots()
+const createRepository = (merkleRoots: MerkleRoot[]): MerkleRootsRepository => {
+  return {
+    saveMerkleRoots: (syncedMerkleRoots: MerkleRoot[]) => {
+      merkleRoots.push(...syncedMerkleRoots);
+      return new Promise((resolve) => {
+        setTimeout(() => resolve(), 50);
```

Comment:
I don't see any reason to simulate timeout of simulated repository in tests.
if we really would like to test such a scenario, then it should be prepared specifically for those tests, but all the others should use simple implementation without timeout so it won't make those test run longer then needed.